mne/export/_edf.py

```diff
@@ -0,0 +1,212 @@
+                               f'is less then physical min passed in {pmin}.')
+
+    # create instance of EDF Writer
+    hdl = EDFwriter(fname, EDFwriter.EDFLIB_FILETYPE_EDFPLUS, n_chs)
+
+    # set channel data
+    for ichan, ch in enumerate(ch_names):
+        if physical_range == 'auto':
+            # take the channel type minimum and maximum
+            ch_type = raw.get_channel_types(picks=ch)[0]
+            pmin, pmax = ch_types_phys_min[ch_type], ch_types_phys_max[ch_type]
+
+        for key, val in zip(
+            ['PhysicalMaximum', 'PhysicalMinimum', 'DigitalMaximum',
+             'DigitalMinimum', 'PhysicalDimension', 'SampleFrequency',
+             'SignalLabel', 'PreFilter',
+             ],
+            [pmax, pmin, digital_max, digital_min, phys_dims, sfreq,
+             ch, filter_str_info]
+        ):
+            _try_to_set_value(hdl, key, val, channel_index=ichan)
+
+    # set patient info
+    subj_info = raw.info.get('subject_info')
+    if subj_info is not None:
+        birthday = subj_info.get('birthday')
+        name = subj_info.get('first_name') + subj_info.get('last_name')
+        hand = subj_info.get('hand')
+        sex = subj_info.get('sex')
+
+        if birthday is not None:
+            if hdl.setPatientBirthDate(birthday[0], birthday[1],
+                                       birthday[2]) != 0:
+                raise RuntimeError(f"Setting Patient Birth Date to {birthday} "
+                                   f"returned an error")
+        for key, val in zip(['PatientName', 'PatientGender',
+                             'AdditionalPatientInfo'],
+                            [name, sex, f'hand={hand}']):
+            _try_to_set_value(hdl, key, val)
+
+    # set measurement date
+    meas_date = raw.info['meas_date']
+    if meas_date:
+        subsecond = meas_date.microsecond / 100.
+        if hdl.setStartDateTime(year=meas_date.year, month=meas_date.month,
+                                day=meas_date.day, hour=meas_date.hour,
+                                minute=meas_date.minute,
+                                second=meas_date.second,
+                                subsecond=subsecond) != 0:  # noqa
+            raise RuntimeError(f"Setting Start Date Time {meas_date} "
+                               f"returned an error")
+
+    device_info = raw.info.get('device_info')
+    if device_info is not None:
+        device_type = device_info.get('type')
+        _try_to_set_value(hdl, 'Equipment', device_type)
+
+    # Write each second (i.e. datarecord) separately.
+    for isec in range(np.ceil(n_secs).astype(int)):
+        end_samp = (isec + 1) * sfreq
+        if end_samp > n_times:
+            end_samp = n_times
+        start_samp = isec * sfreq
+
+        # then for each second write each channel
+        for ich in range(n_chs):
+            # create a buffer with sampling rate
+            buf = np.zeros(sfreq, np.float64, "C")
```

Comment:
I suspect such double for loop for writing will be super slow.... can you profile writting 200MB of data to .edf vs writting to .fiff?

Reply:
That is the limitation of the EDFLib-Python library I believe? because data must be written contiguously as "data records" 

https://github.com/mne-tools/mne-python/pull/9643#issuecomment-893207329

cc: @cbrnr and @Teuniz

Reply:
Just reviving this, but yes I believe forsure writing an EDF file in mne-python will be slower then writing to FIFF